server/src/main/java/io/crate/replication/logical/action/TransportAlterPublicationAction.java

```diff
@@ -164,6 +167,32 @@ static Publication updatePublication(Request request, Metadata currentMetadata,
         return new Publication(oldPublication.owner(), oldPublication.isForAllTables(), new ArrayList<>(tables));
     }
 
+    private static void updateSettings(IndexNameExpressionResolver indexNameExpressionResolver,
+                                       String publicationName,
+                                       Publication oldPublication,
+                                       Publication newPublication,
+                                       ClusterState currentState,
+                                       Metadata.Builder mdBuilder) {
+        // Check for removed tables
+        var oldTables = new ArrayList<>(oldPublication.tables());
+        if (oldTables.removeAll(newPublication.tables())) {
```

Comment:
if it's non-intersecting sets (t1,t2) - old, (t3,t4) - new, `removeAll` would return false but we actually need to apply setting as we removed everything. survived after removeAll is removed elements - so maybe we can change if to `!empty`